src/openapi/transport/retry.js

```diff
@@ -130,29 +139,63 @@ TransportRetry.prototype.sendTransportCall = function(transportCall) {
  * @param transportCall
  */
 TransportRetry.prototype.addFailedCall = function(transportCall) {
-    transportCall.retryCount++;
     this.failedCalls.push(transportCall);
-    if (!this.retryTimer) {
+    const callOptions = this.methods[transportCall.method];
+    if (callOptions.retryTimeouts && callOptions.retryTimeouts.length > transportCall.retryCount) {
+        // schedule an individual retry timeout
+        transportCall.retryTimer = setTimeout(() => {
+            this.retryIndividualFailedCall(transportCall);
+        }, callOptions.retryTimeouts[transportCall.retryCount]);
+    } else if (!this.retryTimer) {
+        // schedule a retry timeout for all failed calls
         this.retryTimer = setTimeout(() => {

```

Comment:
No, it's just to hold all failed/pending calls. My thought was to keep all failed calls in the same array (its name kinda suggested that) and figure out when they fire separately. We better keep the individually timed ones somewhere so we clean up properly, but that could be in a separate array if you think it's more straightforward?